src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -38,10 +42,10 @@ public void OnBeginRequest(HttpContext httpContext, long timestamp)
         {
             if (this.client.IsEnabled())
             {
-                httpContext.Features.Set(new RequestTelemetry());
-
-                this.beginRequestTimestamp.Value = timestamp;
-                this.client.Context.Operation.Id = httpContext.TraceIdentifier;
+                var requestTelemetry = new RequestTelemetry();
+                requestTelemetry.Context.Operation.Id = httpContext.TraceIdentifier;
```

Comment:
I can do that, but I'm unclear what the rules should be here. I had assumed that the original code was correct in setting the operation Id (except that it wasn't correct for simultaneous requests). Should we set both? Or was the original just completely wrong?